src/migrate.ts

```diff
@@ -0,0 +1,27 @@
+import * as S from '@eyedea/syncano'
+
+interface Args {
+  migrationKey: string
+}
+
+class Endpoint extends S.Endpoint {
+  async run({response, users}: S.Core, {args, config}: S.Context<Args>) {
+    if (args.migrationKey === config.CUSTOMERIO_MIGRATION_KEY) {
+      try {
+        const allUsers = await users.list()
+        for (const user of allUsers) {
+          if (user.username && !user.customerId) {
+            users
```

Comment:
You can replace this with mass update:

```
await data.book.update([
  14, {title: 'Book with id 14 was updated'},
  15, {title: 'Book with id 15 was updated'}
])
```

Reply:
It'll make code less readable

Reply:
Performance > Readability

Also it won't be that complicated. 

```jsx
const usernameToCustomerId = item => ([item.id, {customerId: item.username}])
await users.update(allUsers.map(usernameToCustomerId))
```

Reply:
but on the other hand it'll be a single database call - makes sense.